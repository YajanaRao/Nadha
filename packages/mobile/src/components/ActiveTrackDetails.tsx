import React from 'react';
import {View, StyleSheet, Dimensions} from 'react-native';
import FastImage from 'react-native-fast-image';
import {Subheading, Title} from 'react-native-paper';
import {DefaultImage} from './DefaultImage';
import {TrackProps} from '../types';

interface Props {
  track: TrackProps;
}

export const ActiveTrackDetails = ({track}: Props) => {
  return (
    <View style={{flex: 1}}>
      <View style={styles.centerContainer}>
        {track.cover ? (
          <FastImage source={{uri: track.cover}} style={[styles.artCover]} />
        ) : (
          <DefaultImage style={styles.artCover} />
        )}
      </View>
      <View style={styles.centerContainer}>
        <Title numberOfLines={1}>{track.title}</Title>
        <Subheading numberOfLines={1}>
          {track.artist ? track.artist : track.album}
        </Subheading>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  artCover: {
    borderRadius: 12,
    elevation: 4,
    height: Dimensions.get('window').width - 50,
    maxHeight: 300,
    maxWidth: 300,
    width: Dimensions.get('window').width - 50,
  },
  centerContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 16,
    marginVertical: 8,
  },
});
