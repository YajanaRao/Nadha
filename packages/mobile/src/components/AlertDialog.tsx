import React from 'react';
import {Button, Paragraph, Dialog, Portal} from 'react-native-paper';

interface AlertProps {
  visible: boolean;
  title: string;
  message: string;
  action(): void;
  hideDialog(): void;
}

export const AlertDialog = ({
  visible,
  title,
  message,
  action,
  hideDialog,
}: AlertProps) => {
  return (
    <Portal>
      <Dialog visible={visible} onDismiss={hideDialog}>
        <Dialog.Title>{title}</Dialog.Title>
        <Dialog.Content>
          <Paragraph>{message}</Paragraph>
        </Dialog.Content>
        <Dialog.Actions>
          <Button onPress={action}>Ok</Button>
          <Button onPress={hideDialog}>Cancel</Button>
        </Dialog.Actions>
      </Dialog>
    </Portal>
  );
};
