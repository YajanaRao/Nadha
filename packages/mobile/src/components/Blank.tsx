import React from 'react';
import {Button, useTheme} from 'react-native-paper';
import {StyleSheet, View} from 'react-native';
import {Tree} from './Tree';

interface Props {
  text: string;
  fetchData(): void;
}

export const Blank = ({text, fetchData}: Props) => {
  const theme = useTheme();
  const {colors} = theme;
  return (
    <View style={[styles.container, {backgroundColor: colors.background}]}>
      <Tree message={text} />
      <View style={styles.padding}>
        <Button icon="refresh" mode="text" onPress={fetchData}>
          Refresh
        </Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  padding: {
    margin: 4,
  },
});
