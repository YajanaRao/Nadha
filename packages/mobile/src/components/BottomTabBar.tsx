import React from 'react';
import {View, StyleSheet} from 'react-native';
import {Caption, TouchableRipple, Surface, Divider} from 'react-native-paper';
import {PlayerBarContainer} from '../containers/PlayerBarContainer';
import {NavigationState} from '@react-navigation/core';

interface BottomTabBarProps {
  renderIcon(route: NavigationState, focused: boolean): void;
  getLabelText(route: NavigationState): void;
  onTabPress(route: NavigationState): void;
  onTabLongPress(route: NavigationState): void;
  getAccessibilityLabel(route: NavigationState): void;
  state: NavigationState;
}

export const BottomTabBar = ({
  renderIcon,
  getLabelText,
  onTabPress,
  onTabLongPress,
  // getAccessibilityLabel,
  state,
}: BottomTabBarProps) => {
  const {routes, index} = state;
  return (
    <Surface style={{elevation: 4}}>
      <PlayerBarContainer />
      <Divider />
      <Surface style={[styles.container]}>
        {routes.map((route: NavigationState, routeIndex: number) => {
          const isRouteActive = routeIndex === index;
          // const color = isRouteActive ? "green" : "red";
          return (
            <TouchableRipple
              key={route.key}
              style={styles.tabButton}
              // rippleColor={colors.primary}
              // underlayColor={colors.primary}
              borderless
              onPress={() => {
                onTabPress({route});
              }}
              onLongPress={() => {
                onTabLongPress({route});
              }}
              // accessibilityLabel={getAccessibilityLabel({ route })}
            >
              <View style={{justifyContent: 'center', alignItems: 'center'}}>
                {renderIcon({route, focused: isRouteActive})}
                <Caption
                  style={{
                    textAlign: 'center',
                    fontSize: 10,
                    margin: 0,
                    padding: 0,
                  }}>
                  {getLabelText({route})}
                </Caption>
              </View>
            </TouchableRipple>
          );
        })}
      </Surface>
    </Surface>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 4,
  },
  tabButton: {
    flex: 1,
    justifyContent: 'center',
    margin: 0,
    padding: 0,
    alignItems: 'center',
  },
});
