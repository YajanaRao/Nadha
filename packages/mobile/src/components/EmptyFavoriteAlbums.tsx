import React from 'react';
import {Title, Subheading} from 'react-native-paper';
import {View, ViewStyle} from 'react-native';
import {ThemedIcon} from './ThemedIcon';

const CONTAINER: ViewStyle = {
  alignItems: 'center',
  flex: 1,
  justifyContent: 'center',
  margin: 24,
};

export const EmptyFavoriteAlbums = () => {
  return (
    <View style={CONTAINER}>
      <ThemedIcon name="heart-outline" size={80} />
      <Title>No favorites yet.</Title>
      <Subheading style={{textAlign: 'center'}}>
        Songs, albums, artists and playlists you&apos;ve liked will live here
      </Subheading>
    </View>
  );
};
