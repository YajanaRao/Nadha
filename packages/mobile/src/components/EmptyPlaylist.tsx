import React from 'react';
import {Subheading} from 'react-native-paper';
import {View, ViewStyle} from 'react-native';
import {ThemedIcon} from './ThemedIcon';
import {Screen} from './Screen';

const CONTAINER: ViewStyle = {
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  margin: 24,
};

export const EmptyPlaylist = () => {
  return (
    <Screen>
      <View style={CONTAINER}>
        <ThemedIcon name="disc-alert" size={80} />
        <Subheading style={{textAlign: 'center'}}>Empty playlists</Subheading>
      </View>
    </Screen>
  );
};
