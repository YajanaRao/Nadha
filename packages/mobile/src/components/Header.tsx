import React, {useState} from 'react';
import {Searchbar, useTheme} from 'react-native-paper';
import {View} from 'react-native';
import {useDispatch} from 'react-redux';
import {updateQuery} from '../actions/mediaStore';

export const Header = () => {
  const [query, setQuery] = useState('');
  const dispatch = useDispatch();

  const handleChange = (text: string) => {
    setQuery(text);
    dispatch(updateQuery(text));
  };

  const theme = useTheme();
  const {colors} = theme;

  return (
    <View style={{backgroundColor: colors.background}}>
      <Searchbar
        placeholder="Artists, songs or podcasts"
        onChangeText={handleChange}
        value={query}
        icon="magnify"
        style={{margin: 10}}
      />
    </View>
  );
};
