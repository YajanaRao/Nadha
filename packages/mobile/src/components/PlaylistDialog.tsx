import React from 'react';
import {List, Dialog, Portal} from 'react-native-paper';
import {FlatList} from 'react-native';
import {getUserPlaylists} from '../actions/realmAction';

interface Props {
  visible: boolean;
  hideModal(): void;
  addToPlaylist(id: string): void;
}

interface PlaylistProps {
  id: string;
  name: string;
  owner: string;
}

export const PlaylistDialog = ({visible, hideModal, addToPlaylist}: Props) => {
  const data = getUserPlaylists();
  return (
    <Portal>
      <Dialog visible={visible} onDismiss={hideModal}>
        <Dialog.Title style={{textAlign: 'center'}}>
          Add to Playlist
        </Dialog.Title>

        <Dialog.ScrollArea>
          <FlatList
            data={data}
            keyExtractor={(item, index) => index.toString()}
            renderItem={({item}: {item: PlaylistProps}) => (
              <List.Item
                title={item.name}
                description={`by ${item.owner}`}
                left={(props) => <List.Icon {...props} icon="playlist-music" />}
                onPress={() => addToPlaylist(item.id)}
              />
            )}
          />
          {/* </View> */}
        </Dialog.ScrollArea>
      </Dialog>
    </Portal>
  );
};
