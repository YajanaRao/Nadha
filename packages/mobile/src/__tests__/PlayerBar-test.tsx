/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {PlayerBar} from '../components/PlayerBar';

// Note: test renderer must be required after react-native.

it('Playing play bar renders correctly without art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="playing"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});

it('Blank component renders correctly', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  const tree = renderer
    .create(
      <PlayerBar
        status="playing"
        active={track}
        navigateToPlayer={navigation}
        togglePlayback={togglePlayback}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Pause play bar renders correctly without art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="pause"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});

it('Track with album name renders correctly without art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="pause"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});

it('Playing play bar renders correctly with art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="playing"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});

it('Pause play bar renders correctly with art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="pause"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});

it('Track with album name renders correctly with art cover', () => {
  const track = {
    id: '123',
    title: 'title',
    artist: 'Tester',
    cover: 'image',
    path: 'imagepath',
  };
  const navigation = jest.fn();
  const togglePlayback = jest.fn();
  renderer.create(
    <PlayerBar
      status="pause"
      active={track}
      navigateToPlayer={navigation}
      togglePlayback={togglePlayback}
    />,
  );
});
