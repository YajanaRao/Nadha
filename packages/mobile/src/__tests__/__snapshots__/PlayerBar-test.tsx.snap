// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 60,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "elevation": 0,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#d7d1c9",
            "borderRadius": 4,
            "height": 50,
            "width": 50,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "image",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginHorizontal": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 16,
                  "letterSpacing": 0.5,
                  "lineHeight": 24,
                  "marginVertical": 2,
                },
                Object {
                  "margin": 0,
                },
              ],
              Array [
                Object {
                  "fontSize": 16,
                  "letterSpacing": 0.5,
                  "lineHeight": 24,
                  "marginVertical": 2,
                },
                Object {
                  "margin": 0,
                },
              ],
            ],
          ]
        }
      >
        title
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 12,
                  "letterSpacing": 0.4,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "margin": 0,
                },
              ],
              Array [
                Object {
                  "fontSize": 12,
                  "letterSpacing": 0.4,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "margin": 0,
                },
              ],
            ],
          ]
        }
      >
        Tester
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "flex-end",
          "width": 50,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 25.5,
                "height": 51,
                "width": 51,
              },
              undefined,
              Object {
                "margin": 0,
                "padding": 0,
              },
            ],
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 34,
                  "width": 34,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 34,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Ôè§
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
